lib/core/src/persist/sync.rs

```diff
@@ -0,0 +1,651 @@
+            Self::check_commit_update(&tx, &sync_state.record_id, last_commit_time)?;
+        }
+
+        let params = named_params! {
+            ":id": &data.swap_id,
+            ":invoice": &data.invoice,
+            ":preimage": &data.preimage,
+            ":create_response_json": &data.create_response_json,
+            ":claim_fees_sat": &data.claim_fees_sat,
+            ":claim_private_key": &data.claim_private_key,
+            ":payer_amount_sat": &data.payer_amount_sat,
+            ":receiver_amount_sat": &data.receiver_amount_sat,
+            ":mrh_address": &data.mrh_address,
+            ":created_at": &data.created_at,
+            ":payment_hash": &data.payment_hash,
+            ":description": &data.description,
+        };
+        match is_update {
+            true => {
+                tx.execute(
+                    "
+                    UPDATE receive_swaps
+                    SET
+                        invoice = :invoice,
+                        preimage = :preimage,
+                        create_response_json = :create_response_json,
+                        claim_fees_sat = :claim_fees_sat,
+                        claim_private_key = :claim_private_key,
+                        payer_amount_sat = :payer_amount_sat,
+                        receiver_amount_sat = :receiver_amount_sat,
+                        mrh_address = :mrh_address,
+                        created_at = :created_at,
+                        payment_hash = :payment_hash,
+                        description = :description
+                    WHERE id = :id",
+                    params,
+                )?;
+            }
+            false => {
+                tx.execute(
+                    "
+                    INSERT INTO receive_swaps(
+                        id,
+                        invoice,
+                        preimage,
+                        create_response_json,
+                        claim_fees_sat,
+                        claim_private_key,
+                        payer_amount_sat,
+                        receiver_amount_sat,
+                        mrh_address,
+                        created_at,
+                        payment_hash,
+                        description,
+                        state
+                    )
+                    VALUES(
+                        :id,
+                        :invoice,
+                        :preimage,
+                        :create_response_json,
+                        :claim_fees_sat,
+                        :claim_private_key,
+                        :payer_amount_sat,
+                        :receiver_amount_sat,
+                        :mrh_address,
+                        :created_at,
+                        :payment_hash,
+                        :description,
+                        :state
+                    )",
+                    [params, &[(":state", &PaymentState::Created)]]
```

Comment:
We have is_local on the swaps. The main question is do we need to exclude them totally from the update loop (and only count on the onchain data) or should we add them to the update loop but without initiating transactions (claim). What do you think?

Reply:
Currently they are considered in the update loop so state updates are pushed outwards (and merged if necessary), then with either Boltz status/restore they're set to the right final state. May be unnecessary to push outwards and let each instance settle its own final state.

Regarding claiming, I think there was discussion about only letting the initiator to broadcast. I think it's the only non-racy way to do it (since the claim checks I've added assume the chain service already has the other instance's tx indexed, which may or may not be true in very rare edge cases), so if this mitigation is not enough I agree we should just stick to the easy approach.

Another idea could also be to maybe track the claims for a few confirmations to see whether the claim info actually matches, that way if two instances claim we can still settle on which one got confirmed (via script history), but I think that may just be unnecessary overhead, let me know what you think.

Reply:
I'm wondering if we first set them to a Recover state, then after the recovery/chain sync if the swap is ongoing then add it the the update loop for tracking.

Reply:
Best way I see of going about that is to persist them instantly anyway, but as Ross mentioned with a `Recovered`/new state that allows them to be tracked explicitly. It's also easier to debug what's happening that way

Reply:
From testing I think Recover state is helpful. The SDK is behaving differently depending if the rt-sync pull is completed before we start to track swaps. The Recover state will also allow us to start tracking the swap if needed. 

I also see an issue sometimes with the initial chain sync from a fresh state, if the rt-sync pulls after the first chain sync, where the SDK events are suppressed, in the next chain sync all the pulled rt-sync swaps events are emitted.

Reply:
Good points @hydra-yse as Ross mentioned let's add the recovery to the sync in pipeline just before persisting the swaps. Results will be more predictable this way and we will avoid weird behaviors.

Reply:
Done in 79b3d8deb7e130166b6c35a431b02daad85570e8